tensorflow/core/kernels/cwise_op_div_1.cc

```diff
@@ -18,20 +18,19 @@ limitations under the License.
 namespace tensorflow {
 REGISTER6(BinaryOp, CPU, "Div", functor::div, float, Eigen::half, double,
           bfloat16, complex64, complex128);
-REGISTER5(BinaryOp, CPU, "Div", functor::safe_div, uint8, uint16, int16, int32,
-          int64);
-REGISTER5(BinaryOp, CPU, "TruncateDiv", functor::safe_div, uint8, uint16, int16,
+REGISTER6(BinaryOp, CPU, "Div", functor::safe_div, uint8, uint16, int8, int16,
           int32, int64);
+REGISTER6(BinaryOp, CPU, "TruncateDiv", functor::safe_div, uint8, uint16, int8,
+          int16, int32, int64);
 REGISTER6(BinaryOp, CPU, "RealDiv", functor::div, float, Eigen::half, double,
           bfloat16, complex64, complex128);
 REGISTER5(BinaryOp, CPU, "DivNoNan", functor::div_no_nan, Eigen::half, float,
           double, complex64, complex128);
 
 #if GOOGLE_CUDA
-REGISTER9(BinaryOp, GPU, "Div", functor::div, float, Eigen::half, double, uint8,
-          uint16, int16, int64, complex64, complex128);
-REGISTER4(BinaryOp, GPU, "TruncateDiv", functor::div, uint8, uint16, int16,
-          int64);
+REGISTER5(BinaryOp, GPU, "Div", functor::div, float, Eigen::half, double, uint8, uint16);
```

Comment:
Looks like you are removing registrations for "Div"? These have been moved to cwise_op_div_2? Why was that needed? That seems strange: shouldn't we have all the registrations for "Div" together in one file? If we can't have all these registrations in one file, perhaps we should split more evenly (and have a comment explaining why we needed to split)?

Reply:
Hey Josh. Thanks for the review.

I would have kept it in one file, however, the macro for registering this many types no longer exists (REGISTER10), with the removal and basis for using multiple files stemming from this commit: https://github.com/tensorflow/tensorflow/commit/a373b0f6bf3dd3ebd80ad9c72485de9c37c815ae

I have been using https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/cwise_op_mul_1.cc and
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/cwise_op_mul_2.cc as a guideline.

I only split the operation that tried to register 10+ types which was only the GPU 'Div' registration.

Reply:
I much prefer the div operations be in one file. Please split so that e.g. Div & TruncateDiv are in one file and RealDiv and DivNoNan is in the other (or whatever else makes sense given what code is shared between operations).